nsd.c

```diff
@@ -931,6 +931,35 @@ bind8_stats (struct nsd *nsd)
 }
 #endif /* BIND8_STATS */
 
+static
+int cookie_secret_file_read(nsd_type* nsd) {
+	char const* file = nsd->options->cookie_secret_file;
+	assert( file != NULL );
+	FILE* f = fopen(file, "r");
+	/* a non-existing cookie file is not an error */
+	if( f == NULL ) { return errno != EPERM; }
+	/* cookie secret file exists and is readable */
+	nsd->cookie_count = 0;
+	int corrupt = 0;
+	char secret[NSD_COOKIE_SECRET_SIZE * 2 + 2/*'\n' and '\0'*/];
+	for(size_t count = 0; count < NSD_COOKIE_HISTORY_SIZE; count++ ) {
```

Comment:
variable declaration inside for loop is very nice (limiting scope), but a c99 feature, so unfortunately it is not portable :(

Reply:
More occurrences like these above...